frontend/src/metabase-lib/queries/StructuredQuery.ts

```diff
@@ -656,8 +679,10 @@ class StructuredQueryInner extends AtomicQuery {
    * @returns the field options for the provided aggregation
    */
   aggregationFieldOptions(agg: string | AggregationOperator): DimensionOptions {
+    // XXX: We need to fix this function so that it takes expression (custom field) into account
+    // aggregationOperator seem to check if the aggregation would be valid for the the given table
     const aggregation: AggregationOperator =
-      typeof agg === "string" ? this.table().aggregationOperator(agg) : agg;
+      typeof agg === "string" ? this.aggregationOperator(agg) : agg;
```

Comment:
This fixes the issue to take expression into account when calculating the available aggregation operators. By using `StructuredQuery`'s `aggregationOperator` (introduced in this PR) instead of `Table`'s

<img width="384" alt="image" src="https://user-images.githubusercontent.com/1937582/225985893-435026bd-eff5-4623-8818-685c9fa99db5.png">